crypto/keys/multisig/multisig_test.go

```diff
@@ -379,5 +389,41 @@ func TestAminoUnmarshalJSON(t *testing.T) {
 	var pk cryptotypes.PubKey
 	err := cdc.UnmarshalJSON([]byte(pkJSON), &pk)
 	require.NoError(t, err)
-	require.Equal(t, uint32(3), pk.(*kmultisig.LegacyAminoPubKey).Threshold)
+	lpk := pk.(*kmultisig.LegacyAminoPubKey)
+	require.Equal(t, uint32(3), lpk.Threshold)
+	require.Equal(t, 5, len(pk.(*kmultisig.LegacyAminoPubKey).PubKeys))
+
+	for _, key := range pk.(*kmultisig.LegacyAminoPubKey).PubKeys {
+		require.NotNil(t, key)
+		pk := secp256k1.PubKey{}
+		err := pk.Unmarshal(key.Value)
+		require.NoError(t, err)
+	}
+}
+
+func TestProtoMarshalJSON(t *testing.T) {
+	require := require.New(t)
+	pubkeys := generatePubKeys(3)
+	msig := kmultisig.NewLegacyAminoPubKey(2, pubkeys)
+
+	registry := types.NewInterfaceRegistry()
+	cryptocodec.RegisterInterfaces(registry)
+	cdc := codec.NewProtoCodec(registry)
+
+	bz, err := cdc.MarshalInterfaceJSON(msig)
+	require.NoError(err)
+
+	var pk2 cryptotypes.PubKey
+	err = cdc.UnmarshalInterfaceJSON(bz, &pk2)
+	require.NoError(err)
+	require.True(pk2.Equals(msig))
+
+	// Test that we can correctly unmarshal key from keyring output
+	info, err := keyring.NewMultiInfo("my multisig", msig)
+	require.NoError(err)
+
+	ko, err := keyring.MkAccKeyOutput(info)
```

Comment:
This doesn't build as these APIs don't exist in v0.42. @haasted do you know how to address this?

Reply:
https://github.com/cosmos/cosmos-sdk/pull/10043 is my suggestion for a fix for the v0.42.x branch. It does most of what #10061 does, except completely verify the contents of each key. Can that be used?

Reply:
@haasted yes, but please target your PR against this PR - `mergify/bp/release/v0.42.x/pr-10061`. That way we can cleanly merge this one.

Reply:
Everything in #10043 is already included in this PR, see L395. This PR even adds more tests.

I propose to close #10043 and merge this PR. The `MkAccKeyOutput` API doesn't exist in 0.42 (which still uses bech32 for pubkeys), so imo the simplest way is to just remove those lines in the test: https://github.com/cosmos/cosmos-sdk/pull/10113#discussion_r707406324

Does that sound good to you @haasted ?